mashup.js

```diff
@@ -0,0 +1,40 @@
+function instaSearch(button) {
+
+  var searchForm = button.parentElement;
+  var tagname = searchForm.querySelector('input[type="text"]').value;
+  var currentDiv = button.parentElement.parentElement;
+
+  $.ajax({
+    url: 'https://api.instagram.com/v1/tags/'+tagname+'/media/recent?client_id=c1108290c34240808da4c980016c49e1&callback=YOUR-CALLBACK',
+    dataType: 'jsonp',
+    success: function(result){
+      for (var i = 0; i < 1; i++){
+        var url = result.data[i].images.thumbnail.url;
+        var instapic = '<img src="' + url + '"/>';
+
+        $(currentDiv).append(instapic);
+
+      }
+    }
+  });
+}
+
+
+//INSTAGRAM CLIENT REGISTRATION
+//Mashup Project
+//CLIENT INFO
+// CLIENT ID	c1108290c34240808da4c980016c49e1
+// CLIENT SECRET	2d287b8a9ec148afb78395b15ff08bcf
```

Comment:
Careful about posting these anywhere publicly-accessible! You'll want to go in to

http://instagram.com/developer/clients/manage/

and click "Reset secret", so that no one can steal it and pretend to be your app.